crowdfunding/projects/views.py

```diff
@@ -1,3 +1,68 @@
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from .models import Project, Pledge 
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer
+from rest_framework import status, permissions, generics
+from django.http import Http404
+from .permissions import IsOwnerOrReadOnly
+
+
+
+
 
 # Create your views here.
+class PledgeList(generics.ListCreateAPIView):
+    queryset = Pledge.objects.all()
+    serializer_class = PledgeSerializer
+
+    def perform_create(self,serializer):
+        serializer.save(supporter=self.request.user)	
+
+
+class ProjectList(APIView):
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly,IsOwnerOrReadOnly]
+    def get(self, request):
+        projects = Project.objects.all()
+        serializer = ProjectSerializer(projects, many=True)
+        return Response(serializer.data)
+
+    
+    def post(self, request):
+        serializer = ProjectSerializer(data=request.data)
```

Comment:
i'm not sure if this will ever activate your permission_classes as it doesn't seem the step for validating permissions is in here.